cmake/SetupBlaze.cmake

```diff
@@ -19,24 +19,32 @@ set_property(TARGET Blaze PROPERTY
   INTERFACE_INCLUDE_DIRECTORIES ${BLAZE_INCLUDE_DIR})
 set_property(TARGET Blaze PROPERTY
   INTERFACE_LINK_LIBRARIES Lapack)
+target_link_libraries(
+  Blaze
+  INTERFACE
+  Blas
+  GSL::gsl # for BLAS header
+  Lapack
+  )
 
 target_compile_definitions(Blaze
   INTERFACE
-  # Override SMP configurations
+  # Enable external BLAS kernels
+  BLAZE_BLAS_MODE=1
+  # Set default matrix storage order to column-major
+  BLAZE_DEFAULT_STORAGE_ORDER=blaze::columnMajor
```

Comment:
This changes the storage order. Why?

Reply:
Things like partial derivatives, `apply_matrices` etc are implemented for column-major layout. With this config you get column-major matrices from Blaze by default, making it easier to work with those functions. In fact, currently we have code like this in the repo:

```cpp
  // src/Evolution/Systems/NewtonianEuler/Characteristics.cpp
  const Matrix a_x = blaze::DynamicMatrix<double>{
      {0., 1., 0.},
      {b_times_theta - square(u), u * (2. - kappa_over_density),
       kappa_over_density}, ...};
```

Here `Matrix` is col-major and `blaze::DynamicMatrix` is row-major (current default). The storage order could be made explicit here, but switching the default to col-major seems sensible to me to make things consistent. I think we can get rid of the shallow `Matrix` wrapper class very easily with the col-major default.

I split the change into a separate commit.

Reply:
okay, that sounds reasonable. Thanks!